index.js

```diff
@@ -255,8 +255,14 @@ Funnel.prototype.build = function() {
     /*eslint-enable no-lonely-if*/
 
     this._isRebuild = true;
-  } else {
+  } else if (inputPathExists) {
     this.processFilters(inputPath);
+  } else if (!this.allowEmpty) {
+    throw new Error(`You specified a \`"srcDir": ${this.srcDir}\` which does not exist and did not specify \`"allowEmpty": true\`.`);
+  } else { // !inputPathExists && this.allowEmpty
+    // Just make an empty folder so that any downstream consumers who don't know
+    // to ignore this on `allowEmpty` don't get trolled.
+    mkdirp(this.destPath);
```

Comment:
I think we need `mkdirp.sync` here (otherwise we have serious race condition issues...)

```suggestion
    mkdirp.sync(this.destPath);
```